resources/perftest/perftest.cpp

```diff
@@ -0,0 +1,150 @@
+{
+  return std::string("file://myasset_") + std::to_string(i);
+}
+
+///////////////////////////////////////////////////////////
+class OldManagerInterface
+{
+public:
+  OldManagerInterface(const std::unordered_map<std::string, std::string>& refToPrimStrs) : refToPrimStrs{refToPrimStrs} {}
+  std::string resolveEntityReference(const std::string& ref);
+
+private:
+  const std::unordered_map<std::string, std::string>& refToPrimStrs;
+};
+
+std::string OldManagerInterface::resolveEntityReference(const std::string& ref)
+{
+  return refToPrimStrs.at(ref);
+}
+
+///////////////////////////////////////////////////////////
+auto benchmarkOldManager(const std::unordered_map<std::string, std::string>& refToPrimStrs, const std::vector<std::string>& refs)
+{
+  OldManagerInterface oldMgr(refToPrimStrs);
+
+  const auto doLookups = [&oldMgr, &refs]() {
+    for(size_t i = 0; i < nIterations; ++i) {
+      const auto& ref = refs[i];
+      const auto& primStr = oldMgr.resolveEntityReference(ref);
+      if(primStr.empty()) {
+        std::cerr << "resolveEntityReference() incorrect result!" << std::endl;
+        exit(-1);
+      }
+    }
+  };
+
+  return MeasureRuntimeMs(doLookups);
+}
+
+///////////////////////////////////////////////////////////
+class NewManagerInterface
+{
+public:
+  NewManagerInterface(const std::unordered_map<std::string, std::string>& refToPrimStrs) : refToPrimStrs{refToPrimStrs} {}
+  std::shared_ptr<specification::Specification> resolve(const std::string& ref, const specification::Specification::TraitIds& traitIds);
+
+private:
+  const std::unordered_map<std::string, std::string>& refToPrimStrs;
+};
+
+std::shared_ptr<specification::Specification> NewManagerInterface::resolve(const std::string& ref, const specification::Specification::TraitIds& traitIds)
+{
+  specification::Specification::TraitIds populatedTraits;
+  bool getPath = false;
+  if(std::find(traitIds.cbegin(), traitIds.cend(), trait::BlobTrait::kId) != traitIds.cend()) {
+    populatedTraits.emplace_back(trait::BlobTrait::kId);
+    getPath = true;
+  }
+  auto result = std::make_shared<specification::Specification>(populatedTraits);
+  if(getPath) {
+    trait::BlobTrait btrait(result);
+    btrait.setUrl(refToPrimStrs.at(ref));
+  }
+  return result;
+}
+
+///////////////////////////////////////////////////////////
+auto benchmarkNewManager(const std::unordered_map<std::string, std::string>& refToPrimStrs, const std::vector<std::string>& refs)
+{
+  NewManagerInterface newMgr(refToPrimStrs);
+
+  const auto doLookups = [&newMgr, &refs]() {
+    for(size_t i = 0; i < nIterations; ++i) {
+      const auto& ref = refs[i];
+      const auto spec = newMgr.resolve(ref, {trait::BlobTrait::kId});
+      trait::BlobTrait btrait(spec);
+      trait::property::Str url;
+      const auto success = btrait.getUrl(&url);
+      if(success != trait::TraitPropertyStatus::kFound) {
+        std::cerr << "getUrl() failed!" << std::endl;
+        exit(-1);
```

Comment:
[consider] It's a bit odd to exit with a negative status, at least on Linux where I believe the program exit code will be `255` if you use `-1`.  A more portable way would be to use `EXIT_FAILURE`.

Reply:
Similar comment for `benchmarkOldManager`.

Reply:
Though do we even need to check the error case, which should never happen (or rather if it does we'll get an exception from `refToPrimStrs.at(` which will terminate the program anyway).

Reply:
Removed error check in https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/270/commits/a1c5a4cbb5b29080ff35a32873be0290a921c406
I think we were worried the compiler might optimize away the code somehow but that's probably not an issue. Quick check shows same timing.